rasa/core/training/interactive.py

```diff
@@ -919,43 +908,35 @@ def _intents_from_messages(messages):
 
 async def _correct_wrong_nlu(
     corrected_nlu: Dict[Text, Any],
-    evts: List[Dict[Text, Any]],
+    events: List[Dict[Text, Any]],
     endpoint: EndpointConfig,
     sender_id: Text,
 ) -> None:
     """A wrong NLU prediction got corrected, update core's tracker."""
 
-    latest_message = latest_user_message(evts)
-    corrected_events = all_events_before_latest_user_msg(evts)
-
-    latest_message["parse_data"] = corrected_nlu
-
-    await replace_events(endpoint, sender_id, corrected_events)
-
-    await send_message(
+    revert_latest_user_utterance = UserUtteranceReverted().as_dict()
+    listen_for_next_message = ActionExecuted(ACTION_LISTEN_NAME).as_dict()
+    corrected_message = latest_user_message(events)
+    corrected_message["parse_data"] = corrected_nlu
+    await send_event(
         endpoint,
         sender_id,
-        latest_message.get("text"),
-        latest_message.get("parse_data"),
+        [revert_latest_user_utterance, listen_for_next_message, corrected_message],
     )
 
 
 async def _correct_wrong_action(
     corrected_action: Text,
     endpoint: EndpointConfig,
     sender_id: Text,
-    finetune: bool = False,
     is_new_action: bool = False,
 ) -> None:
     """A wrong action prediction got corrected, update core's tracker."""
 
-    result = await send_action(
+    _ = await send_action(
```

Comment:
```suggestion
    await send_action(
```

Reply:
Mhm, I did it so it's clear that I am omitting a result there. But since I didn't find clear guidelines for it, I will revert it 👍